src/datascience-ui/interactive-common/textAreaFocusTracker.tsx

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+import * as React from 'react';
+
+export class TextAreaFocusTracker extends React.Component {
+    private lastFocusedTextArea: HTMLElement | undefined;
+    public componentDidMount() {
+        window.addEventListener('focus', () => this.onWindowGotFocus(), false);
+        window.addEventListener('blur', () => this.onWindowLostFocus(), false);
+    }
+    public render() {
+        const hiddenStyle: React.CSSProperties = {
+            display: 'none'
+        };
+
+        return <div id="focus_tracker" style={hiddenStyle} />;
+    }
+
+    private onWindowGotFocus() {
+        // When the window receives focus (from outside), on a delay attempt to reset our focus
+        setTimeout(this.setToLastKnown.bind(this), 100);
```

Comment:
So this mostly seems ok. But the timeout still seems a bit dangerous to me. Would this possibly rip focus back if you set focus elsewhere during the timeout? 100ms is unlikely, but could be possible. Would it make sense to cancel the timeout in onWindowLostFocus? I think that we would not want to run it if focus had been removed in that time.

Reply:
Cancel sounds good. I don't think 100ms is necessary either. I can make it zero.